### PR TITLE
Helm service resiliency

### PR DESCRIPTION
**Pull request by BerniVarga**

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -112,31 +112,30 @@ func runServer(neoURL string, port string, cacheDuration string, env string) {
...
-
-	http.Handle("/", router())
+func routeRequests(port string) {
 
-	if err := http.ListenAndServe(":"+port, nil); err != nil {
-		log.Fatalf("Unable to start server: %v", err)
+	// Standard endpoints
+	healthCheck := fthealth.TimedHealthCheck{
+		HealthCheck: fthealth.HealthCheck{
+			SystemCode:  "up-csa",
```
Comment (tosan88) [inline app.go]: I think these fields are wrongly populated

Comment (BerniVarga) [reply app.go]: Updated! Thanks :)

File: annotations/health_handler.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package annotations
+
+import (
+	fthealth "github.com/Financial-Times/go-fthealth/v1_1"
+	"github.com/Financial-Times/service-status-go/gtg"
+)
+
+const (
+	deweyUrl = "https://dewey.ft.com/up-csa.html"
```
Comment (peteclark-ft) [inline annotations/health_handler.go]: Should be public-annotations-api

File: annotations/health_handler.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		BusinessImpact:   "Unable to respond to Public Annotations api requests",
+		Name:             "Check connectivity to Neo4j",
+		PanicGuide:       deweyUrl,
+		Severity:         1,
+		TechnicalSummary: `Cannot connect to Neo4j. If this check fails, check that Neo4j instance is up and running. You can find the neoUrl as a parameter for this service.`,
+		Checker:          Neo4jChecker,
+	}
+}
+
+func Neo4jChecker() (string, error) {
```
Comment (peteclark-ft) [inline annotations/health_handler.go]: Minor, but I'd usually code a function like this to return the happy path on the last line of the function i.e.:
```
err := AnnotationsDriver.checkConnectivity()
if err != nil {
   return "Error connecting to neo4j", err   
}
return "Connectivity to neo4j is ok", nil
```

File: annotations/health_handler.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	return "Error connecting to neo4j", err
+}
+
+func GoodToGo() gtg.Status {
+	statusCheck := func() gtg.Status {
+		return gtgCheck(Neo4jChecker)
+	}
+
+	return gtg.FailFastParallelCheck([]gtg.StatusChecker{statusCheck})()
```
Comment (peteclark-ft) [inline annotations/health_handler.go]: Do we need a `FailFastParallelCheck` for a single gtg check?

File: app.go [hunk 5]
```diff
@@ -112,31 +112,30 @@ func runServer(neoURL string, port string, cacheDuration string, env string) {
...
+func routeRequests(port string) {
 
-	if err := http.ListenAndServe(":"+port, nil); err != nil {
-		log.Fatalf("Unable to start server: %v", err)
+	// Standard endpoints
+	healthCheck := fthealth.TimedHealthCheck{
+		HealthCheck: fthealth.HealthCheck{
+			SystemCode:  "public-annotations-api",
+			Name:        "Public Annotations API Healthcheck",
+			Description: "Checking Neo4j",
```
Comment (peteclark-ft) [inline app.go]: I think this should be the app description?

File: annotations/health_handler.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package annotations
+
+import (
+	fthealth "github.com/Financial-Times/go-fthealth/v1_1"
+	"github.com/Financial-Times/service-status-go/gtg"
+)
+
+const (
+	deweyUrl = "https://dewey.in.ft.com/view/system/up-nvam"
```
Comment (peteclark-ft) [inline annotations/health_handler.go]: Isn't this next annotations video mapper? I think this should be https://dewey.in.ft.com/view/system/annotationsapi

Comment (BerniVarga) [reply annotations/health_handler.go]: I'm so distracted. You are right. I'm going to change it in a minute.

**Pull request merged**